fabric-rendering-v1/src/main/java/net/fabricmc/fabric/api/client/rendering/v1/BuiltinItemRendererRegistry.java

```diff
@@ -42,6 +47,43 @@
 	 * @param renderer the renderer
 	 * @throws IllegalArgumentException if the item already has a registered renderer
 	 * @throws NullPointerException if either the item or the renderer is null
+	 * @deprecated Please use {@link BuiltinItemRendererRegistry#register(Item, DynamicItemRenderer)} instead.
 	 */
+	@Deprecated
 	void register(Item item, BuiltinItemRenderer renderer);
+
+	/**
+	 * Registers the renderer for the item.
+	 *
+	 * <p>Note that the item's JSON model must also extend {@code minecraft:builtin/entity}.
+	 *
+	 * @param item     the item
+	 * @param renderer the renderer
+	 * @throws IllegalArgumentException if the item already has a registered renderer
+	 * @throws NullPointerException if either the item or the renderer is null
+	 */
+	void register(ItemConvertible item, DynamicItemRenderer renderer);
+
+	/**
+	 * Dynamic item renderers render items with custom code.
+	 * They allow using non-model rendering, such as BERs, for items.
+	 *
+	 * <p>An item with a dynamic renderer must have a model extending {@code minecraft:builtin/entity}.
+	 * The renderers are registered with {@link BuiltinItemRendererRegistry#register(Item, DynamicItemRenderer)}.
+	 */
+	@FunctionalInterface
+	@Environment(EnvType.CLIENT)
+	interface DynamicItemRenderer {
```

Comment:
I think this would be better as a top-level type like the old interface.

Reply:
Given this name, it's better left at top level indeed. :+1: